drivers/clock_control/clock_control_bl60x.c

```diff
@@ -0,0 +1,942 @@
+		      ((uint32_t)(1) << PDS_CLKPLL_INT_FRAC_SW_POS);
+	} else {
+		tmp = (tmp & PDS_CLKPLL_ICP_1U_UMSK) | ((uint32_t)(0) << PDS_CLKPLL_ICP_1U_POS);
+		tmp = (tmp & PDS_CLKPLL_ICP_5U_UMSK) | ((uint32_t)(2) << PDS_CLKPLL_ICP_5U_POS);
+		tmp = (tmp & PDS_CLKPLL_INT_FRAC_SW_UMSK) |
+		      ((uint32_t)(0) << PDS_CLKPLL_INT_FRAC_SW_POS);
+	}
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_CP_OFFSET);
+
+	/* More 26M special treatment */
+	tmp = sys_read32(PDS_BASE + PDS_CLKPLL_RZ_OFFSET);
+	if (clock_control_bl60x_crystal_to_id(crystal_frequency) == 4) {
+		tmp = (tmp & PDS_CLKPLL_C3_UMSK) | ((uint32_t)(2) << PDS_CLKPLL_C3_POS);
+		tmp = (tmp & PDS_CLKPLL_CZ_UMSK) | ((uint32_t)(2) << PDS_CLKPLL_CZ_POS);
+		tmp = (tmp & PDS_CLKPLL_RZ_UMSK) | ((uint32_t)(5) << PDS_CLKPLL_RZ_POS);
+		tmp = (tmp & PDS_CLKPLL_R4_SHORT_UMSK) | ((uint32_t)(0) << PDS_CLKPLL_R4_SHORT_POS);
+	} else {
+		tmp = (tmp & PDS_CLKPLL_C3_UMSK) | ((uint32_t)(3) << PDS_CLKPLL_C3_POS);
+		tmp = (tmp & PDS_CLKPLL_CZ_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_CZ_POS);
+		tmp = (tmp & PDS_CLKPLL_RZ_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_RZ_POS);
+		tmp = (tmp & PDS_CLKPLL_R4_SHORT_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_R4_SHORT_POS);
+	}
+	tmp = (tmp & PDS_CLKPLL_R4_UMSK) | ((uint32_t)(2) << PDS_CLKPLL_R4_POS);
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_RZ_OFFSET);
+
+	/* set pll dividers */
+	tmp = sys_read32(PDS_BASE + PDS_CLKPLL_TOP_CTRL_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_POSTDIV_UMSK) | ((uint32_t)(0x14) << PDS_CLKPLL_POSTDIV_POS);
+	tmp = (tmp & PDS_CLKPLL_REFDIV_RATIO_UMSK) | ((uint32_t)(2) << PDS_CLKPLL_REFDIV_RATIO_POS);
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_TOP_CTRL_OFFSET);
+
+	/* set SDMIN */
+	tmp = sys_read32(PDS_BASE + PDS_CLKPLL_SDM_OFFSET);
+	if (source == BL60X_CLKID_CLK_CRYSTAL) {
+		tmp = (tmp & PDS_CLKPLL_SDMIN_UMSK) |
+		      (clock_control_bl60x_crystal_SDMIN_table[clock_control_bl60x_crystal_to_id(
+			       crystal_frequency)]
+		       << PDS_CLKPLL_SDMIN_POS);
+	} else {
+		tmp = (tmp & PDS_CLKPLL_SDMIN_UMSK) |
+		      (clock_control_bl60x_crystal_SDMIN_table[clock_control_bl60x_crystal_to_id(
+			       RC32M_FREQ)]
+		       << PDS_CLKPLL_SDMIN_POS);
+	}
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_SDM_OFFSET);
+
+	/* phase comparator settings? */
+	tmp = sys_read32(PDS_BASE + PDS_CLKPLL_FBDV_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_SEL_FB_CLK_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_SEL_FB_CLK_POS);
+	tmp = (tmp & PDS_CLKPLL_SEL_SAMPLE_CLK_UMSK) |
+	      ((uint32_t)(1) << PDS_CLKPLL_SEL_SAMPLE_CLK_POS);
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_FBDV_OFFSET);
+
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_PU_CLKPLL_SFREG_UMSK) | ((uint32_t)(1) << PDS_PU_CLKPLL_SFREG_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	clock_control_bl60x_clock_settle();
+
+	/* enable PPL clock actual? */
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_PU_CLKPLL_UMSK) | ((uint32_t)(1) << PDS_PU_CLKPLL_POS);
```

Comment:
I think we should update all similar ones in this file
```suggestion
	tmp = (tmp & PDS_PU_CLKPLL_UMSK) | (1U << PDS_PU_CLKPLL_POS);
```

Reply:
done